manual/src/cmds/runtime-tracing.etex

```diff
@@ -70,6 +74,31 @@ The full set of events emitted by probes and their documentation can be found in
  section~\ref{Runtime_events}.
 \fi
 
+\item[Custom probes] are defined using the
+\texttt{Runtime_events.User} module.
+Identified by a globally unique name, they record events and provide payloads of
+built-in (\texttt{unit}, \texttt{int}, \texttt{span}) and user-defined types.
+Consumers and producers are not compiled together, so they don't usually have
+the same set of registered events. The event name is used by the consumer to
+identify producer's events, with a fallback mechanism for unknown events of
+built-in types. Unknown events for user-defined types are currently ignored,
+even if the type is defined both in the producer and consumer side.
```

Comment:
You explain "runtime probes" and "custom probes", but in fact most of the explanation of "custom probes" is about what "custom events" are. My feeling is that "events" and "probes" are two different concepts that could be explained separately: events have a semantic and a payload, and probes, actually I am not sure what they are (the Runtime_events module documentation does not contain the word "probe"), but they seem to be related to both emission and reception of events -- or at least both are discussed here.

Reply:
You're right. Basically I see probes as "the piece of code that emit events". There are probes in the runtime that emit runtime events. Using the custom events library, users can set up additional probes in their code base that emit custom events. 

I'll come up with better wording.

Reply:
Note (not directly related to the PR): to me the word "probe" suggests a tool to observe some information, so it is more associated to a "receiver" than to a "producer" in my mind. The usage of the word "probe" in a similar context that I know of are the probes in the Linux kernel ( https://docs.kernel.org/trace/kprobes.html ), and their meaning there is fairly different from "the piece of code that emit events" (the probe is more like the part that registers a listener for a given event, the "probed instructrion" is in charge of emitting the event).

Reply:
The other precedent for the use of the term 'probe' is the dtrace, which supports both dynamic and static probe points. https://illumos.org/books/dtrace/chp-usdt.html (certificate expired, but content still valid)

Reply:
I see yes, my context for the definition of "probe" is:
- https://github.com/janestreet/ocaml-probes: "Some examples of what a probe handler can do: write data into a ring buffer or similar for transmission to a another tool that can analyze or visualize it.", a 'probe handler' being "If a probe is enabled, whenever the execution reaches it, the corresponding handler is evaluated."
- the current usage of probe in the manual: "The full set of events emitted by probes [...]" 

Both making me think of them as event producers.

Reply:
(cc @sadiqj if you have opinions on how "probes" should be defined, and how they relate to the documentation the Runtime_events module itself.)

Reply:
If you're into electronics, a probe is part of the measurement device (oscilloscope, etc).  The part of the printed circuit board where you can attach a probe is called a test point, or _point de piquage_ in French.  What about "tap" as a source of events?  Or just "event source" for additional clarity.

Reply:
I don't think I have a strong opinion either way on terminology. I used to refer to the CAML_EV macros as probes because there wasn't an explicit way of actually attaching to those, if runtime_events was enabled all probes produced events that went in to the ringbuffer - you didn't really attach to individual ones directly.

Reply:
In any case I think that the discussion in this paragraph mixes two many things: (1) what is the information carried by those custom events, (2) how to emit them, (3) how to listen for them. I think that it would be natural to discuss at least (1) strictly before (2), (3), which could be discussed either together or separately.

(I think it would also be nice to understand what terminology we want to use, but this is orthogonal. I do find it weird that currently the terminology in the manual chapter is distinct from the terminology in the .mli documents.)